compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -386,15 +387,21 @@ impl Instruction {
             | IfElse { .. }
             | ArraySet { .. } => true,
 
+            // Store instructions must be removed by DIE in acir code, any load
+            // instructions should already be unused by that point.
+            Store { .. } => {
+                matches!(function.runtime(), RuntimeType::Acir(_))
+                    && function.reachable_blocks().len() == 1
+            }
+
```

Comment:
This is assuming some knowledge about the significance of only having a single block on the part of the reader. Can you flesh this out for the benefits of auditors?